basetest.pm

```diff
@@ -328,7 +330,8 @@ sub runtest {
     };
     if ($@) {
         # copy the exception early
-        my $internal = Exception::Class->caught('OpenQA::Exception::InternalException');
+        my $exception = Exception::Class->caught('OpenQA::Exception::InternalException');
+        my $internal = blessed $exception && !$exception->isa('OpenQA::Exception::FailedNeedle');
```

Comment:
hmm, why is FailedNeedle an InternalException. I think that's the oversight here

Reply:
Well, it has been like this since commit 1759b5e5 with the description
    
    For specific exceptions (like Needle Failed) we don't record an extra
    test result, only for exceptions thrown unexpectedly

so could you think of why you did it like this?